app/com/yetu/oauth2provider/controllers/OAuth2Auth.scala

```diff
@@ -86,26 +89,35 @@ class OAuth2Auth(authorizationHandler: handlers.AuthorizationHandler,
           val authorizeRequest = authClient.request
 
           if (client.coreYetuClient) {
-            authorizeService.handlePermittedApps(client, authorizeRequest, request.user)
+            Future.successful(authorizeService.handlePermittedClient(client, authorizeRequest, request.user))
           } else {
-            authorizeService.handleClientPermissions(request, env, client, authorizeRequest, request.user)
+            authorizeService.handleNonCoreClient(request, env, client, authorizeRequest, request.user)
```

Comment:
Not knowing the API this might be extraneous but it looks strange to me that on the if branch we are returning a Future and on the else branch we are not.